src/dungeon_data.c

```diff
@@ -539,4 +539,38 @@ void init_dungeons(void)
     }
 }
 
+void recount_creatures_and_diggers()
+{
+    struct Dungeon* dungeon;
+    unsigned short i;
+    for (i = 0; i < PLAYERS_COUNT; i++)
+    {
+        dungeon = get_dungeon(i);
+        if (!dungeon_invalid(dungeon))
+        {
+            dungeon->num_active_creatrs = 0;
+            dungeon->num_active_diggers = 0;
+        }
+    }
+    for (i = 0; i < THINGS_COUNT; i++)
```

Comment:
the proper way to loop trough creatures would be 
get_list_for_thing_class 
...
thing->next_of_class
this avoids needlessly checking all things that aren't creatures

Reply:
We need a way to clear and reset the creature and digger lists too.